helix-view/src/clipboard.rs

```diff
@@ -110,9 +112,17 @@ pub fn get_clipboard_provider() -> Box<dyn ClipboardProvider> {
             copy => "termux-clipboard-set";
         }
     } else if env_var_is_set("TMUX") && exists("tmux") {
-        command_provider! {
-            paste => "tmux", "save-buffer", "-";
-            copy => "tmux", "load-buffer", "-";
+        if config.tmux_system_clipboard {
+            command_provider! {
+                // Refresh tmux clipboard, wait a bit for it to be updated and paste it
+                paste => "sh", "-c", "tmux refresh-client -l; sleep 0.1; tmux save-buffer -";
+                copy => "tmux", "load-buffer", "-w", "-";
+            }
```

Comment:
How about adding a `editor.clipboard.paste`/`editor.clipboard.copy` (now that I think about it, `copy` -> `yank`) section to the configuration? That way you're able to set this in `config.toml`

Reply:
Hm, ideally I’d like the clipboard to just work out of the box. (Apart from this PR, there’s also OSC 52 support missing (which are terminal escape sequences), they are needed to support copy and paste e.g. in alacritty through ssh.)

How do you propose should the config look like if these commands should be used inside tmux, but the other built-in commands should be used outside tmux, e.g. when DISPLAY is set.

Reply:
If that config is enabled but tmux is not enabled won't it have an issue?

Reply:
You mean with the `tmux-system-clipboard` from this PR? That shouldn’t be a problem. It’s only used when helix is running inside tmux.

With `editor.clipboard.paste` commands set in the config that are executed regardless of the environment that helix runs in, that would be a problem.